src/utils/logging.ts

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export interface LogInterface {
+    debug(msg: string, ...supportingData: any[]): void;
+    warn(msg: string, ...supportingData: any[]): void;
+    error(msg: string, ...supportingData: any[]): void;
+    info(msg: string, ...supportingData: any[]): void;
+}
+
+export enum LogLevel {
+    debug,
+    info,
+    warn,
+    error
+}
+
+
+export class Log implements LogInterface {
```

Comment:
Let's call this class `ConsoleLogger` (because it logs to the console). Then we can call the interface `Logging`.

Reply:
Done